src/Integration.UnitTests/Binding/CSharpVbBindingConfigTests.cs

```diff
@@ -36,41 +38,60 @@ public class CSharpVbBindingConfigTests
-            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("filePath");
+             act = () => new CSharpVBBindingConfig(ruleSet, null);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("additionalFile");
+
+            act = () => new CSharpVBBindingConfig(ruleSet, additionalFile, null);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("fileSystem");
         }
 
         [TestMethod]
-        public void GetSolutionLevelFilePaths_ReturnPathToRulesetFile()
+        public void GetSolutionLevelFilePaths_ReturnFilePaths()
         {
-            var testSubject = new CSharpVBBindingConfig(new RuleSet("dummy"), "c:\\test.txt");
-            testSubject.SolutionLevelFilePaths.Count().Should().Be(1);
-            testSubject.SolutionLevelFilePaths.First().Should().Be(testSubject.FilePath);
+            var ruleSet = new FilePathAndContent<RuleSet>("ruleset dummy", new RuleSet("dummy"));
+            var additionalFile = new FilePathAndContent<SonarLintConfiguration>("additional file dummy", new SonarLintConfiguration());
+
+            var testSubject = new CSharpVBBindingConfig(ruleSet, additionalFile);
+            testSubject.SolutionLevelFilePaths.Count().Should().Be(2);
+            testSubject.SolutionLevelFilePaths.First().Should().Be(ruleSet.Path);
+            testSubject.SolutionLevelFilePaths.Last().Should().Be(additionalFile.Path);
         }
 
         [TestMethod]
-        public void Save_ValidFilePath_SaveCalled()
+        public void Save_FilesSaved()
         {
             // We can't mock the RuleSet class so we're testing Save by actually
             // writing to disk.
             // Arrange
             var testDir = Path.Combine(TestContext.DeploymentDirectory, TestContext.TestName);
             Directory.CreateDirectory(testDir);
-            var fullPath = Path.Combine(testDir, "savedRuleSet.txt");
 
-            var testSubject = new CSharpVBBindingConfig(new RuleSet("dummy"), fullPath);
+            var rulesetFullPath = Path.Combine(testDir, "savedRuleSet.txt");
+            var additionalFileFullPath = Path.Combine(testDir, "additionalFile.txt");
+
+            var ruleSet = new FilePathAndContent<RuleSet>(rulesetFullPath, new RuleSet("dummy"));
+            var additionalFile = new FilePathAndContent<SonarLintConfiguration>(additionalFileFullPath, new SonarLintConfiguration());
+
+            var testSubject = new CSharpVBBindingConfig(ruleSet, additionalFile);
 
             // Act
             testSubject.Save();
 
             // Assert
-            File.Exists(fullPath).Should().BeTrue();
+            File.Exists(rulesetFullPath).Should().BeTrue();
+            File.Exists(additionalFileFullPath).Should().BeTrue();
+
+            var savedAdditionalFile = File.ReadAllText(additionalFileFullPath);
+            savedAdditionalFile.Should().Be(@"{
+  ""Settings"": [],
+  ""Rules"": []
```

Comment:
I didn't notice this when looking at the product code, but the `SonarLint.xml` file should be saved as XML rather than json...

It makes me wonder if there's anything else I've missed.

Reply:
I pushed a fix 👍